src/com/tekion/dtos/Match.java

```diff
@@ -0,0 +1,176 @@
+package com.tekion.dtos;
+
+import java.util.Scanner;
+import java.util.Random;
+import com.tekion.enums.*;
+import com.tekion.validator.UserInputValidator;
+
+public class Match {
+    private int overs;
+    private Team team1 , team2;
+    private Team teamBattingFirst , teamFieldingFirst;
+    private int targetScore = 0;
+
+    public void playMatch(){
+        this.setupMatch();
+        this.setTeamInfo();
+/*
+      WEEK-2
+      this.showTeamInfo();
+*/
+        this.coinToss();
+        this.playFirstInning();
+        this.inningsBreak();
+        this.playSecondInning();
+        this.displayResult();
+    }
+
+    private void setupMatch(){
+        Scanner sc = new Scanner(System.in);
+        System.out.println("Please choose match type (T20/0DI): ");
+        String userInput = sc.nextLine();
+//        while(!UserInputValidator.matchInputValidator(userInput)){
+//            System.out.println("Incorrect Match type. Please type again.");
+//            userInput = sc.nextLine();
+//        }
+        try {
+            TypesOfMatch matchType = TypesOfMatch.valueOf(userInput.toUpperCase());
```

Comment:
MatchType
we need to focus on variable naming. it is very important in the long run maintainability of code. spend time thinking of variable names. 
Questions to ask : 
1. Are they concise enough?
2. Can they be shortened further?
3. Are the names appropriate

Variable naming is an acquired skill. it requires a lot of practice but once used to it, it becomes very beneficial